src/Language/PureScript/Ide.hs

```diff
@@ -116,130 +112,84 @@ findPursuitPackages :: (MonadIO m) =>
-  case getDependenciesForModule <$> m of
-    Just deps -> do
-      mapM_ loadModule deps
-      -- We need to load the modules, that get reexported from the dependencies
-      depModules <- catMaybes <$> mapM getModule deps
-      -- What to do with errors here? This basically means a reexported dependency
-      -- doesn't exist in the output/ folder
-      traverse_ loadReexports depModules
-      pure ("Dependencies for " <> moduleName <> " loaded.")
-    Nothing -> throwError (ModuleNotFound moduleName)
-
-loadReexports :: (PscIde m, MonadLogger m, MonadError PscIdeError m) =>
-                Module -> m [ModuleIdent]
-loadReexports m = case getReexports m of
-  [] -> pure []
-  exportDeps -> do
-    -- I'm fine with this crashing on a failed pattern match.
-    -- If this ever fails I'll need to look at GADTs
-    let reexports = map (\(Export mn) -> mn) exportDeps
-    $(logDebug) ("Loading reexports for module: " <> fst m <>
-                 " reexports: " <> T.intercalate ", " reexports)
-    traverse_ loadModule reexports
-    exportDepsModules <- catMaybes <$> traverse getModule reexports
-    exportDepDeps <- traverse loadReexports exportDepsModules
-    return $ concat exportDepDeps
-
-getDependenciesForModule :: Module -> [ModuleIdent]
-getDependenciesForModule (_, decls) = mapMaybe getDependencyName decls
-  where getDependencyName (Dependency dependencyName _ _) = Just dependencyName
-        getDependencyName _ = Nothing
-
-loadModule :: (PscIde m, MonadLogger m, MonadError PscIdeError m) =>
-              ModuleIdent -> m Text
-loadModule "Prim" = pure "Prim won't be loaded"
-loadModule mn = do
-  path <- filePathFromModule mn
-  loadExtern path
-  $(logDebug) ("Loaded extern file at: " <> T.pack path)
-  pure ("Loaded extern file at: " <> T.pack path)
-
-loadAllModules :: (PscIde m, MonadLogger m, MonadError PscIdeError m) => m Success
+-- | Takes the output directory and a filepath like "Monad.Control.Eff" and
+-- looks up, whether that folder contains an externs.json
+checkExternsPath :: FilePath -> FilePath -> IO (Maybe FilePath)
+checkExternsPath oDir d
+  | d `elem` [".", ".."] = pure Nothing
+  | otherwise = do
+      let file = oDir </> d </> "externs.json"
+      ex <- doesFileExist file
+      if ex
+        then pure (Just file)
+        else pure Nothing
+
+findAllExterns :: (Ide m, MonadError PscIdeError m) => m [FilePath]
+findAllExterns = do
+  oDir <- outputDirectory
+  unlessM (liftIO (doesDirectoryExist oDir))
+    (throwError (GeneralError "Couldn't locate your output directory."))
+  liftIO $ do
+    dirs <- getDirectoryContents oDir
+    externPaths <- traverse (checkExternsPath oDir) dirs
+    pure (catMaybes externPaths)
+
+loadModules
+  :: (Ide m, MonadError PscIdeError m, MonadLogger m)
+  => [P.ModuleName]
+  -> m Success
+loadModules mns = do
+  oDir <- outputDirectory
+  let efPaths = map (\mn -> oDir </> P.runModuleName mn </> "externs.json") mns
+  efiles <- traverse readExternFile efPaths
```

Comment:
Could you please add some line comments here explaining what's going on, in stages?